backend/src/Types/Api.hs

```diff
@@ -27,10 +28,37 @@ data GameReport = GameReport
   }
   deriving (Generic)
 
-instance FromJSON GameReport
+instance FromJSON RawGameReport
+
+instance ToJSON RawGameReport
+
+toGameReport :: UTCTime -> PlayerId -> PlayerId -> RawGameReport -> GameReport
+toGameReport timestamp winnerId loserId r =
+  GameReport
+    { gameReportTimestamp = timestamp,
+      gameReportWinnerId = winnerId,
```

Comment:
Does the `gameReport` prefix magically map to the `game_report` table when this is run through `SQL.insert`?? So confused but this is cool haha

Reply:
It's dark magic indeed, made darker by the fact that the TH obscures the real types/kinds that are going on behind the scenes. But the gist is this: the tables and the entities that populate them are compile-time related by their types. When you say `SQL.insert someGameReport`, you're not just inserting a blob of arbitrary data into a specified table... you're inserting a strongly typed record that can *only* be inserted into the table it fits in - anything else would be a type error. And because of that, type inference works everything out for us. No need to specify a table or anything. Each generated record type has one and only one table where it fits.

Reply:
That is freakin cool. And it would be so hard for a runtime bug to get through all that. `Each generated record type has one and only one table where it fits` - perfection